src/liteclient.cc

```diff
@@ -243,25 +243,16 @@ bool LiteClient::finalizeInstall(data::InstallationResult* ir) {
   return ret.result_code.num_code == data::ResultCode::Numeric::kOk;
 }
 
-Uptane::Target LiteClient::getRollbackTarget(bool ignore_ostree_hash, bool allow_current) {
-  const auto rollback_hash{ignore_ostree_hash ? ""
-                                              : sysroot_->getDeploymentHash(OSTree::Sysroot::Deployment::kRollback)};
-
-  Uptane::Target rollback_target{Uptane::Target::Unknown()};
-  {
-    std::vector<Uptane::Target> installed_versions;
-    storage->loadPrimaryInstallationLog(
-        &installed_versions, true /* make sure that Target has been successfully installed */, allow_current);
-
-    std::vector<Uptane::Target>::reverse_iterator it;
-    for (it = installed_versions.rbegin(); it != installed_versions.rend(); it++) {
-      if (ignore_ostree_hash || it->sha256Hash() == rollback_hash) {
-        rollback_target = *it;
-        break;
-      }
-    }
+Uptane::Target LiteClient::getRollbackTarget(bool allow_current) {
+  std::vector<Uptane::Target> installed_versions;
+  storage->loadPrimaryInstallationLog(&installed_versions,
+                                      true /* make sure that Target has been successfully installed */, allow_current);
+  if (installed_versions.empty()) {
+    LOG_WARNING << "No fully installed target found to rollback to";
```

Comment:
I think we need to tackle this case next or at least create a jira ticket.

Reply:
I've created a ticket for that.